miasm2/arch/aarch64/arch.py

```diff
@@ -0,0 +1,1944 @@
+    parser = base_expr
+
+
+class aarch64_imm_64(aarch64_imm_32):
+    parser = base_expr
+
+
+class aarch64_int64_noarg(int32_noarg):
+    parser = base_expr
+    intsize = 64
+    intmask = (1 << intsize) - 1
+    int2expr = lambda self, x: m2_expr.ExprInt64(
+        sign_ext(x, self.l, self.intsize))
+
+
+class aarch64_uint64_noarg(imm_noarg):
+    parser = base_expr
+    intsize = 64
+    intmask = (1 << intsize) - 1
+    int2expr = lambda self, x: m2_expr.ExprInt64(x)
+
+
+class aarch64_uint64(aarch64_uint64_noarg, m_arg):
+    parser = base_expr
+
+
+def set_imm_to_size(size, expr):
+    if size == expr.size:
+        return expr
+    if size > expr.size:
+        expr = m2_expr.ExprInt_fromsize(size, expr.arg)
+    else:
+        if expr.arg > (1 << size) - 1:
+            return None
+        expr = m2_expr.ExprInt_fromsize(size, expr.arg)
+    return expr
+
+
+class aarch64_imm_sf(imm_noarg):
+    parser = base_expr
+
+    def fromstring(self, s, parser_result=None):
+        start, stop = super(aarch64_imm_sf, self).fromstring(s, parser_result)
+        if start is None:
+            return start, stop
+        size = self.parent.args[0].expr.size
+        if self.expr in gpregs64_info.expr + gpregs32_info.expr:
+            return None, None
+        if isinstance(self.expr, m2_expr.ExprOp):
+            return False
+        expr = set_imm_to_size(size, self.expr)
+        if expr is None:
+            return None, None
+        self.expr = expr
+        return start, stop
+
+    def encode(self):
+        if not isinstance(self.expr, m2_expr.ExprInt):
+            return False
+        if not test_set_sf(self.parent, self.expr.size):
+            return False
+        value = int(self.expr.arg)
+        if value >= 1 << self.l:
+            return False
+        self.value = value
+        return True
+
+    def decode(self, v):
+        size = 64 if self.parent.sf.value else 32
+        self.expr = m2_expr.ExprInt_fromsize(size, v)
+        return True
+
+
+class aarch64_imm_sft(aarch64_imm_sf, m_arg):
+
+    def encode(self):
+        if not isinstance(self.expr, m2_expr.ExprInt):
+            return False
+        if not test_set_sf(self.parent, self.expr.size):
+            return False
+        value = int(self.expr.arg)
+        if value < 1 << self.l:
+            self.parent.shift.value = 0
+        else:
+            if value & 0xFFF:
+                return False
+            value >>= 12
+            if value >= 1 << self.l:
+                return False
+            self.parent.shift.value = 1
+        self.value = value
+        return True
+
+    def decode(self, v):
+        size = 64 if self.parent.sf.value else 32
+        if self.parent.shift.value == 0:
+            self.expr = m2_expr.ExprInt_fromsize(size, v)
+        elif self.parent.shift.value == 1:
+            self.expr = m2_expr.ExprInt_fromsize(size, v << 12)
+        else:
+            return False
+        return True
+
+option2size = [32, 32, 32, 64,
```

Comment:
Sorry for not quoting every vars to rewrote as upper case. This is a remark which is apply-able on others file of this commit, and for `option2size`, `ext2_*`